Dockerfile

```diff
@@ -21,23 +25,32 @@ RUN apk add --update --no-cache \
     fribidi-dev \
     libimagequant-dev \
     libxcb-dev libpng-dev \
-    && pip install --upgrade pip setuptools\
-    && pip install --no-cache-dir -r requirements.txt
+    gcc \
+    libffi-dev \
+    musl-dev \
+    curl \
+    && pip install --upgrade --no-cache-dir pip==23.3.2 \
+    && curl -sSL https://install.python-poetry.org | python3 - \
+    && /root/.local/bin/poetry self add poetry-plugin-export
 
-# Installing Pillow separate from the packages in requirements
-# greatly speeds up the docker build.
-RUN python3 -m pip install --upgrade pip \
-    && python3 -m pip install Pillow==10.2.0 --global-option="build_ext" --global-option="--disable-tiff" --global-option="--disable-freetype" --global-option="--disable-lcms" --global-option="--disable-webp" --global-option="--disable-webpmux" --global-option="--disable-imagequant" --global-option="--disable-xcb"
 
-FROM bitnami/kubectl:1.28.2 as kubectl
-FROM alpine/helm:3.12.3 as helm
 
-# Non-root user with sudo access
-FROM python:3.8-alpine3.17 as build
-COPY --from=base /usr/local/lib/python3.8/site-packages/ /usr/local/lib/python3.8/site-packages/
-COPY --from=base /usr/local/bin/ /usr/local/bin/
-COPY --from=kubectl /opt/bitnami/kubectl/bin/kubectl /usr/local/bin/
-COPY --from=helm /usr/bin/helm /usr/local/bin/
+# If build-args is set to DISABLE_EXTRA=true, then we skip all superfluous software
+RUN if [ "$DISABLE_EXTRAS" = "true" ]; then \
+    /root/.local/bin/poetry export -f requirements.txt --output requirements.txt; \
+    else /root/.local/bin/poetry export --all-extras  -f requirements.txt --output requirements.txt; \
+    fi
+
+RUN pip install --no-cache-dir -r requirements.txt
```

Comment:
```suggestion
RUN poetry config virtualenvs.create false --global
# If build-args is set to DISABLE_EXTRA=true, then we skip all superfluous software
RUN if [ "$DISABLE_EXTRAS" = "true" ]; then \
    /root/.local/bin/poetry install -n -q --no-cache --only main; \
    else /root/.local/bin/poetry install -n -q --no-cache --all-extras; \
    fi
```

I think `--only main` is the right option

Reply:
I have fixed this now. The solution was close to this, but not quite.